### DISCUSSION TITLE
shut down

### DISCUSSION THREAD

**Discussion by Senthiele** [jschuh/klipper-macros#121] [Q&A] [answered]
ok I keep getting this error. any idea on how to fix this?

7:48 AM
Shutdown due to required printer.cfg section(s) missing:
7:48 AM
Shutdown due to required printer.cfg section(s) missing:
Missing CANCEL_PRINT in on_error_gcode for [virtual_sdcard] section.
See readme: https://github.com/jschuh/klipper-macros#klipper-setup
Once the underlying issue is corrected, use the
"FIRMWARE_RESTART" command to reset the firmware, reload the
config, and restart the host software.
Printer is shutdown

**Comment by jschuh** [comment, marked as answer]
The link in the error message points to the config portion of the readme. I've pasted the exact section below, with the line you're missing in bold:

`[virtual_sdcard]`
`path: ~/gcode_files # UPDATE THIS FOR YOUR PATH!!!`
**`on_error_gcode: CANCEL_PRINT`**

That's actually been required pretty much forever to make cancellation work correctly, but I noticed only recently that it was missing from the documentation and config check macro.

**Reply by Senthiele** [reply]
so do i put this in the readme file or my print.cfg file

**Reply by jschuh** [reply]
If you followed [the installation instructions](https://github.com/jschuh/klipper-macros#installation) you should already have a `[virtual_sdcard]` section in your `printer.cfg` (or in a file you `include`'d from your `printer.cfg`). You just need to add the `on_error_gcode: CANCEL_PRINT` line to that `[virtual_sdcard]` section.

The [readme](https://github.com/jschuh/klipper-macros/blob/main/README.md) is just the documentation for these macros (including installation instructions). And as with any other files from this repo, your configuration should not involve modifying it.

**Reply by Senthiele** [reply]
ok ty
